### PR TITLE
Feature/mac client

### PR DESCRIPTION
**Pull request by metelik**
Support for MAC addresses EUI 48 and EUI 64 in the DHCP option 61 (client_id)

### CODE REVIEW

**Comment by metelik** [comment]
I could have done the checks with a single regex BUT in suhc a way the intentions are much clearer. Moreover such a parsing does not require superb efficiency as it is going to be performed e'er an' anon mostly after applying the new configuration.

File: lib/nerves_network/dhclientv4_conf.ex [hunk 1]
```diff
@@ -79,8 +87,42 @@ defmodule Nerves.Network.Dhclientv4Conf do
...
+   field.  The value in UC_Len_i does not include the length field
+   itself and MUST be non-zero.  Let m be the number of User Classes
+   carried in the option.  The length of the option as specified in Len
+   must be the sum of the lengths of each of the class names plus m:
+   Len= UC_Len_1 + UC_Len_2 + ... + UC_Len_m + m.  If any instances of
+   User Class Data are present, the minimum value of Len is two (Len =
+   UC_Len_1 + 1 = 1 + 1 = 2).
+
+   The Code for this option is 77.
+  """
```
Comment (michaelkschmidt) [inline lib/nerves_network/dhclientv4_conf.ex:122]: Well done with the docs for this!!

File: lib/nerves_network/dhclientv4_conf.ex [hunk 2]
```diff
@@ -147,21 +186,22 @@ defmodule Nerves.Network.Dhclientv4Conf do
     interface "<%= @interface %>" {\n\
     <%= if @host_name do %>  send host-name "<%= @host_name %>";\n<% end %>\
     <%= if @vendor_class_identifier do %>  send vendor-class-identifier "<%= @vendor_class_identifier %>";\n<% end %>\
+    <%= if @hardware_type do %>\
+    <%= if @client_identifier do %>  send dhcp-client-identifier <%= @client_identifier %>;\n<% end %>\
+    <% else %>\
     <%= if @client_identifier do %>  send dhcp-client-identifier "<%= @client_identifier %>";\n<% end %>\
+    <% end %>\
```
Comment (michaelkschmidt) [inline lib/nerves_network/dhclientv4_conf.ex:193]: I struggled with extra newlines in other EEx templates.  That is awesome you figured out the "\" trick

**Review by michaelkschmidt — APPROVED** [review]
Looks good!

File: lib/nerves_network/dhclientv4_conf.ex (hunk 2 above)
Comment (metelik) [reply lib/nerves_network/dhclientv4_conf.ex:193]: :)

Comment (metelik) [reply lib/nerves_network/dhclientv4_conf.ex:193]: 'twas drivin' me nuts :)

**Pull request merged**